IceCream/Classes/DatabaseManagerProtocol.swift

```diff
@@ -125,24 +125,40 @@ extension DatabaseManager {
                     self.syncRecordsToCloudKit(recordsToStore: recordsToStore, recordIDsToDelete: recordIDsToDelete, completion: completion)
                 }
             case .chunk:
-                /// CloudKit says maximum number of items in a single request is 400.
-                /// So I think 300 should be fine by them.
-                print("chunking - records to store : \(recordsToStore.count), records to delete : \(recordIDsToDelete.count)")
-                let chunkedToStoreRecords = recordsToStore.chunkItUp(by: 300)
-                for chunk in chunkedToStoreRecords {
-                    if chunk.count > 0 {
-                        self.syncRecordsToCloudKit(recordsToStore: chunk, recordIDsToDelete: [], completion: completion)
+
+                // chunk up the request and use dispatch groups to execute completion once all chunked requests have finished
+                let dispatchGroup = DispatchGroup()
+                var lastError : Error? = nil
+                
+                let chunkCompletion : ((Error?) -> ())? = { error in
+                    if let error = error {
+                        print("^^ error while chunking \(error)")
```

Comment:
is there any other logging that needs to happen here?

Reply:
@jacobbullock i added some logging that got removed accidentally.  let me know if you want to log anything else.  thanks!

Reply:
@premedrchang I was actually referring to capturing those logs in firebase or something. Do you want to do anything like that or is the print statement fine?

Reply:
@jacobbullock yeah... since this is a forked repo, i am a little hesitant to add new logging dependencies... but we've already been updating a fair amount of logic, maybe we just take full ownership?  then we can add PixiteLogging aka `plog(.warn, ...)` etc and it'll log to our app's file logs.... what do you think?

Reply:
I think logging to the app logs is great if we can do it, but I wouldn't hold up this PR for that.